index.js

```diff
@@ -161,9 +174,23 @@ tslintPlugin.report = function(reporter, options) {
     var throwErrors = function() {
         // Throw error
         if (options && options.emitError === true && errorFiles.length > 0) {
-            return this.emit('error', new PluginError('gulp-tslint', 'Failed to lint: ' + allFailures.map(function(failure) {
+            var failuresToOutput = allFailures;
+            var ignoreFailureCount = 0;
+
+            if (options.reportLimit > 0) {
+                var ignoreFailureStart = options.reportLimit;
+                ignoreFailureCount = allFailures.length - options.reportLimit;
+
+                failuresToOutput = allFailures.splice(ignoreFailureStart, ignoreFailureCount)
```

Comment:
Why do you output the "ignored" errors? If there are, say, 10000 ignored errors and reportLimit is 100, then the thrown error will include 9900 errors. splice doesn't return the modified array, it will modify the allFailures array here, then return a new array of the deleted elements.

It would make more sense to use Array.prototype.slice on allFailures. It would keep the allFailures array intact for possible other uses, and the thrown error messge will include the same errors that possibly were reported.

Something like
```
failuresToOutput = allFailures.slice(0, options.reportLimit);
```
should work here.